### PR TITLE
Added Resnet9 architecture and data prefetch loader

### PR DESCRIPTION
**Pull request by ravinkohli**
Added resnet9 architecture for fast training on cifar10, and data prefetching capability. Also, made changes in the search space for network selector which was causing .refit to not work on networks other than resnet.

### CODE REVIEW

File: autoPyTorch/core/autonet_classes/autonet_image_data.py [hunk 1]
```diff
@@ -93,9 +94,10 @@ def _apply_default_pipeline_settings(pipeline):
         net_selector.add_network('resnet152', ResNet152)
         net_selector.add_network('darts', DARTSImageNet)
         net_selector.add_network('mobilenet', MobileNet)
-        net_selector._apply_search_space_update('resnet:nr_main_blocks', [2, 4], log=False)
-        net_selector._apply_search_space_update('resnet:widen_factor_1', [0.5, 8], log=True)
-
+        net_selector.add_network('resnet9', ResNet9)
+        # net_selector._apply_search_space_update('resnet:nr_main_blocks', [2, 4], log=False)
```
Comment (LMZimmer) [inline autoPyTorch/core/autonet_classes/autonet_image_data.py:98]: Should these be here or not?

File: autoPyTorch/core/autonet_classes/autonet_image_data.py [hunk 2]
```diff
@@ -113,10 +115,10 @@ def _apply_default_pipeline_settings(pipeline):
...
         lr_selector.add_lr_scheduler('none', SchedulerNone)
         
-        lr_selector._apply_search_space_update('step:step_size', [1, 100], log=True)
-        lr_selector._apply_search_space_update('step:gamma', [0.001, 0.99], log=True)
-        lr_selector._apply_search_space_update('cosine_annealing_with_restarts:T_max', [1, 100], log=True)
-        lr_selector._apply_search_space_update('cosine_annealing_with_restarts:T_mult', [1., 2.], log=False)
+        # lr_selector._apply_search_space_update('step:step_size', [1, 100], log=True)
+        # lr_selector._apply_search_space_update('step:gamma', [0.001, 0.99], log=True)
+        # lr_selector._apply_search_space_update('cosine_annealing_with_restarts:T_max', [1, 100], log=True)
+        # lr_selector._apply_search_space_update('cosine_annealing_with_restarts:T_mult', [1., 2.], log=False)
```
Comment (LMZimmer) [inline autoPyTorch/core/autonet_classes/autonet_image_data.py:121]: Again, should those be commented out?

File: autoPyTorch/core/autonet_classes/autonet_image_data.py (hunk 1 above)
Comment (ravinkohli) [reply autoPyTorch/core/autonet_classes/autonet_image_data.py:98]: These search space updates cause error when I tried running any network other than resnet. Without these lines, these hyperparameters are still in the resnet search space as they have been added to the get_configspace method of the model.

File: autoPyTorch/core/autonet_classes/autonet_image_data.py (hunk 2 above)
Comment (ravinkohli) [reply autoPyTorch/core/autonet_classes/autonet_image_data.py:121]: Similar reason to the above

File: autoPyTorch/core/autonet_classes/autonet_image_data.py (hunk 1 above)
Comment (ravinkohli) [reply autoPyTorch/core/autonet_classes/autonet_image_data.py:98]: These search space updates cause error when I tried running any network other than resnet. Without these lines, these hyperparameters are still in the resnet search space as they have been added to the get_configspace method of the model.

File: autoPyTorch/core/autonet_classes/autonet_image_data.py (hunk 2 above)
Comment (ravinkohli) [reply autoPyTorch/core/autonet_classes/autonet_image_data.py:121]: Similar reason to above.

Comment (LMZimmer) [reply autoPyTorch/core/autonet_classes/autonet_image_data.py:121]: Could you please add the default values [here](https://github.com/automl/Auto-PyTorch/blob/2fb1fb77cab28127aed04d19b4a79ecf028f1c30/autoPyTorch/components/lr_scheduler/lr_schedulers.py#L289)?

Comment (ravinkohli) [reply autoPyTorch/core/autonet_classes/autonet_image_data.py]: I have pushed these changes, the PR is also updated.

**Review by LMZimmer — COMMENTED** [review]
`my_job.sh` is an unnecessary file in this PR. Everything else is ready to merge

**Comment by ravinkohli** [comment]
> `my_job.sh` is an unnecessary file in this PR. Everything else is ready to merge

I've deleted it

**Pull request closed without merging**